delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/indexing/IndexingStreamBehaviour.scala

```diff
@@ -0,0 +1,284 @@
+      id: Iri,
+      fetchView: (Iri, ProjectRef) => UIO[Option[ViewIndex[V]]],
+      buildStream: IndexingStream[V],
+      clearIndex: ClearIndex,
+      projection: Projection[Unit],
+      retryStrategy: RetryStrategy[Throwable]
+  )(implicit uuidF: UUIDF, sc: Scheduler): Behavior[IndexingViewCommand[V]] =
+    Behaviors.setup[IndexingViewCommand[V]] { context =>
+      Behaviors.withStash(100) { buffer =>
+        import context._
+
+        def streamName(rev: Long) = s"${project}_${id}_$rev"
+
+        // When the stream completes or receives an error, we stop the actor
+        def onFinalize(uuid: UUID) = UIO.delay(self ! StreamStopped(uuid))
+
+        def runStream(current: Option[(ViewIndex[V], StreamSwitch)]): Unit = {
+          val io: Task[IndexingViewCommand[V]] = fetchView(id, project).flatMap {
+            case None                                    =>
+              UIO.delay(
+                logger.info("View {} in project {} can't be found, the indexing will be passivated", id, project)
+              ) >> current.fold(Task.unit)(_._2.stop).as(ViewNotFound)
+            case Some(fetchView) if fetchView.deprecated =>
+              UIO.delay(
+                logger.info("View {} in project {} is deprecated, the indexing will be passivated", id, project)
+              ) >> current.fold(Task.unit)(_._2.stop).as(ViewIsDeprecated)
+            case Some(fetchView)                         =>
+              for {
+                progress <- projection.progress(fetchView.projectionId)
+                switch   <-
+                  current match {
+                    case None                                                             =>
+                      UIO.delay(
+                        logger
+                          .debug(
+                            "Index view {} with revision {} in project {} will start from offset {}",
+                            id,
+                            fetchView.rev,
+                            project,
+                            progress.offset
+                          )
+                      ) >>
+                        StreamSwitch.run(
+                          streamName(fetchView.rev),
+                          buildStream(fetchView, progress),
+                          retryStrategy,
+                          onFinalize,
+                          onFinalize
+                        )
+                    case Some((currentIndex, switch)) if fetchView.rev > currentIndex.rev =>
+                      UIO.delay(
+                        logger.debug(
+                          "New revision of view {} in project {} is detected, the indexing will be restart from the beginning",
+                          fetchView.rev,
+                          id,
+                          project
+                        )
+                      ) >>
+                        switch.stop >> clearIndex(currentIndex.index) >> restartStream(fetchView)
```

Comment:
It just should be an UIO and not a Task no ?

Reply:
Yeah probably that. But on the caller side we should at least log the error. It is not critical if we don't delete the previous index but it can create a lot of garbage, so we would like to know that happened.

Alternatively we could have a healing process that deletes unused indices (in the future).

Reply:
I just logged it for now, I created an issue to discuss about setting up a healing process: https://github.com/BlueBrain/nexus/issues/2176